nexoan/crud-api/db/queryExecutor/Interfaces/executor.go

```diff
@@ -0,0 +1,454 @@
+	//           "kind": "Person",
+	//           "department": "Engineering",
+	//       },
+	//   })
+	GraphEntityFilter(ctx context.Context, plan *QueryParameter) ([]map[string]interface{}, error)
+
+	// GraphRelationships retrieves all relationships for an entity.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing entity identifier
+	//
+	// Returns:
+	//   - map[string]*pb.Relationship: Map of relationships keyed by relationship ID
+	//   - error: Any error that occurred during retrieval
+	//
+	// Example:
+	//   relationships, err := executor.GraphRelationships(ctx, &QueryParameter{
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//       },
+	//   })
+	GraphRelationships(ctx context.Context, plan *QueryParameter) (map[string]*pb.Relationship, error)
+
+	// FilteredRelationships retrieves relationships based on filter criteria.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing filter criteria
+	//
+	// Returns:
+	//   - map[string]*pb.Relationship: Map of filtered relationships
+	//   - error: Any error that occurred during filtering
+	//
+	// Example:
+	//   relationships, err := executor.FilteredRelationships(ctx, &QueryParameter{
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//           "relationshipType": "MANAGES",
+	//           "direction": "OUTGOING",
+	//       },
+	//       TimeRange: struct{
+	//           ActiveAt: "2024-03-21T00:00:00Z",
+	//       },
+	//   })
+	FilteredRelationships(ctx context.Context, plan *QueryParameter) (map[string]*pb.Relationship, error)
+
+	// RelationshipsCreate creates new relationships for an entity.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing relationship details
+	//
+	// Returns:
+	//   - error: Any error that occurred during creation
+	//
+	// Example:
+	//   err := executor.RelationshipsCreate(ctx, &QueryParameter{
+	//       Operation: "create_relationship",
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//           "relatedEntityId": "456",
+	//           "relationshipType": "MANAGES",
+	//           "startTime": "2024-01-01T00:00:00Z",
+	//       },
+	//   })
+	RelationshipsCreate(ctx context.Context, plan *QueryParameter) error
+
+	// RelationshipsUpdate updates existing relationships.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing update details
+	//
+	// Returns:
+	//   - error: Any error that occurred during update
+	//
+	// Example:
+	//   err := executor.RelationshipsUpdate(ctx, &QueryParameter{
+	//       Operation: "update_relationship",
+	//       Filters: map[string]interface{}{
+	//           "relationshipId": "rel123",
+	//           "endTime": "2024-03-21T00:00:00Z",
+	//       },
+	//   })
+	RelationshipsUpdate(ctx context.Context, plan *QueryParameter) error
```

Comment:
```suggestion
	UpdateRelationships(ctx context.Context, plan *QueryParameter) error
```

I am re-thinking here, should we use operator name before and put the operand.